Lib/fontTools/feaLib/ast.py

```diff
@@ -179,13 +179,20 @@ class Expression(Element):
 class Comment(Element):
     """A comment in a feature file."""
 
-    def __init__(self, text, location=None):
+    def __init__(self, text, location=None, raw=False):
         super(Comment, self).__init__(location)
+        self.raw = not text.strip().startswith("#")
```

Comment:
how about you check at contruction time whether the text starts with `#` and if it doesn't you add it at the beginning of each line. basically move the logic currently in asFea to the `__init__`. I would prefer that to having this stateful self.raw. A comment must always start with `# ` by definition; we allow to omit it to make it easier to code with but internally it has (must have) the leading `#` otherwise it isn't a comment.